Upgrade the Apache Maven wrapper to 3.6.0

Building the SCT master using the current maven wrapper 3.3.3 leads to checkstyle error like this:
```[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:check (no-http-checkstyle-validation) on project spring-cloud-task-parent: Unable to parse configuration of mojo org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:check for parameter sourceDirectories: Cannot assign configuration entry 'sourceDirectories' with value './' of type java.lang.String to property of type java.util.List -> [Help 1]```

The 3.3.3 is rather old maven version

--- a/.mvn/wrapper/maven-wrapper.properties
+++ b/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/batch-events/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/batch-events/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/batch-job/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/batch-job/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/jpa-sample/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/jpa-sample/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/multiple-datasources/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/multiple-datasources/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/partitioned-batch-job/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/partitioned-batch-job/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/task-events/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/task-events/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/taskprocessor/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/taskprocessor/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-samples/tasksink/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-samples/tasksink/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

--- a/spring-cloud-task-stream/.mvn/wrapper/maven-wrapper.properties
+++ b/spring-cloud-task-stream/.mvn/wrapper/maven-wrapper.properties
@@ -1,1 +1,1 @@
-distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip

